rllib/evaluation/postprocessing.py

```diff
@@ -178,7 +178,24 @@ def compute_gae_for_sample_batch(
         input_dict = sample_batch.get_single_step_input_dict(
             policy.model.view_requirements, index="last"
         )
-        last_r = policy._value(**input_dict)
+
+        if policy.config["_enable_rl_module_api"]:
+            # TODO (Kourosh) This is a hack which I don't like at all. We need a
+            # permanent solution.
+            # Note: During sampling you are using the parameters at the beginning of
+            # the sampling process. If I'll be using this advantages during training
```

Comment:
Are you referring to the off-policy'ness for e.g. PPO minibatch SGD iterations?

Otherwise (if num_sgd_iter=1), it wouldn't matter that much, correct? B/c we only calculate advantages for on-policy algos.

Reply:
yes, if you have num_sgd_iter > 1 then you are off-policy on even advantage computations. You'll use the same advantages as num_sgd_iter=1.